### ISSUE TITLE
FR: native sidechain effect which applies a sidechain style volume curve to the audio routed through it

### ISSUE THREAD

**Issue by dmt-hub** [mixxxdj/mixxx#13987]
### Feature Description

<img width="363" alt="Screenshot 2024-12-05 at 16 47 46" src="https://github.com/user-attachments/assets/a024df5e-ccee-44ea-815d-7a5099684b4a">
Would be great to have a native mixxx effect which applies this volume (kick to sidechain-like) curve to the audio.
with various timing option 1/1  1/2  1/4  1/8  1/16

**Comment by Swiftb0y** [comment]
You can achieve something similar with the already built in `Tremolo` effect. The main difference is that it currently only has a very limited envelope/waveform that doesn't match the kick-ducking curve as well as the one you posted.

**Comment by fwcd** [comment]
Not exactly what you're asking for, but I think it would also be cool to have support for "real" sidechain compression, i.e. sidechaining one deck to a compressor assigned to another one.

**Comment by Swiftb0y** [comment]
real sidechain will be much more difficult though code-wise. just fyi.

**Comment by Swiftb0y** [comment]
I think its important to remember that mixxx is not a DAW but a realtime DJ programm. Sometimes simplicity trumps power when it comes to easy of use and especially how quickly it can be used. I don't think its super realistic to expect the user to design or even tweak the ducking curve during their set.

**Comment by fwcd** [comment]
Yes and no, I've got to admit that I find the possibilities that are opening up with Mixxx getting more and more DAW-style features quite intriguing. In this specific case, maybe the user wouldn't want to tweak the curve live, but might already have a few good presets. Essentially it would improve the ability to "remix" tracks on-the-fly and as long as we don't burden regular users with these features, having more stuff under the hood for power users is always nice.

Personally I think these things are good candidates for plugins and prioritizing plugin integrations is likely easier than integrating these effects (along with GUIs) natively, but in the long-term I don't see any harm with adding more specific enveloping tools too.

**Comment by dmt-hub** [comment]
Even a basic envelope ducking with that one curve (non tweakable) would be a nice addition and def very useable very a regular dj to duck the kick in tracks. I noticed Djay Pro has this effect available too, and its great to have.. But i prefer using Mixxx 🗡️
